spec/spec-schema-provider-discoverability.md

```
@@ -0,0 +1,810 @@
+
+#### Automatic Discovery (Zero Configuration)
+```properties
+# No schema provider configuration needed - automatic SPI discovery
+pi2schema.personal.metadata.provider=pi2schema.schema.providers.jsonschema.personaldata.JsonSchemaPersonalMetadataProvider
+```
+
+### Practical Pi2Schema Module Structure
+
+Based on the current pi2schema project structure, here's how the modules would be organized:
+
+```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)